### DISCUSSION TITLE
Using max threads for schema and index creation

### DISCUSSION THREAD

**Discussion by sqgly** [mydumper/mydumper#1220] [Q&A]
Hi all, 

I am running a myloader process with 16vcpu on AWS ECS. I am also using the very latest release 0.14.5-3.

I seem to lose some speed during the constraint, schema and index creation. So I tried to boost the max threads for these steps.

My settings look like this. I am also using what is in master branch for mydumper.cnf file

```
myloader --host="xxx" --user="xxx" --password="xxx" --source-db="xxx" --database="xxx" \
            --compress-protocol \
            --skip-post \
            --skip-definer \
            --skip-triggers \
            --innodb-optimize-keys \
            --verbose=3 \
            --threads=24 \
            --directory=/xxx/backup \
            --defaults-file=mydumper.cnf \
            --max-threads-for-index-creation=6 \
            --max-threads-for-schema-creation=6  \
            --max-threads-per-table=16  
```

I consistently face this error   `Segmentation fault      (core dumped)` when using max threads for schema and index.

Any advice?

Thanks!

**Comment by sqgly** [comment]
As a side note, @davidducos is there anything that stands out to you as an obvious error to cause the loading to be slow? 
Dealing with ~500GB of data.

**Comment by davidducos** [comment]
Not sure about the seg fault issue, it might be fixed on the latest pre release which you can test and let me know if you see the same behavior

**Reply by sqgly** [reply]
Thanks @davidducos I will test it and let you know
